File: getsurei.cs
/* Author: nakinor
 * Created: 2012-03-13
 * Revised: 2012-03-13
 */

// Visual C# による月齢表示プログラム(コンソール)

using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace getsurei
{
    class Program
    {
        static void Main(string[] args)
        {

            int[] a_array = {0, 2, 0, 2, 2, 4, 5, 6, 7, 8, 9, 10};
            int a = 0, b = 0, c = 0;
            int getsurei = 0;
            DateTime dt= DateTime.Now;
            // dt に入れた日時は明示的に int 設定しなくてもいいのかな？

            a = ((dt.Year - 11) % 19) * 11;
            b = a_array[dt.Month - 1]; // Month は 1 から 12 を返す。
            c = dt.Day;
            getsurei = (a + b + c) % 30;

            Console.Write("今日は{0}年{1}月{2}日です。\n",
                           dt.Year, dt.Month, dt.Day);
            Console.Write("月齢は約{0}日です。\n", getsurei);

        }
    }
}

File: getsurei.R
# Author: nakinor
# Created: 2017-12-20
# Revised: 2017-12-20

# 月齢を計算して表示するプログラム

# 今日
today <- Sys.Date()
year <- format(today, "%Y")
month <- format(today, "%m")
day <- format(today, "%d")

# 配列を用意 (引数は 1 からはじまる)
arr <- c(0, 2, 0, 2, 2, 4, 5, 6, 7, 8, 9, 10)

# 計算用の値
a <- ((as.integer(year) - 11) %% 19) * 11
b <- arr[as.integer(month)]
c <- as.integer(day)

# 計算 (剰余は % が 2 つ)
getsurei <- (a + b + c) %% 30

# 出力 (これなんとかならん？)
cat("今日は")
cat(year)
cat("年")
cat(month)
cat("月")
cat(day)
cat("日です。")
cat("月齢は約")
cat(getsurei)
cat("日です。\n")

#cat("今日は", year, "年", month, "月", day, "日です。")
#cat("月齢は約", getsurei, "日です。\n")
